lib/remote/apilistener-configsync.cpp

```diff
@@ -454,19 +502,17 @@ void ApiListener::SendRuntimeConfigObjects(const JsonRpcConnection::Ptr& aclient
 	Log(LogInformation, "ApiListener")
 		<< "Syncing runtime objects to endpoint '" << endpoint->GetName() << "'.";
 
-	for (const Type::Ptr& type : Type::GetAllTypes()) {
-		auto *dtype = dynamic_cast<ConfigType *>(type.get());
-
-		if (!dtype)
-			continue;
-
-		for (const ConfigObject::Ptr& object : dtype->GetObjects()) {
-			/* don't sync objects for non-matching parent-child zones */
-			if (!azone->CanAccessObject(object))
-				continue;
-
-			/* send the config object to the connected client */
-			UpdateConfigObject(object, nullptr, aclient);
+	std::unordered_set<ConfigObject*> syncedObjects;
+	for (const Type::Ptr& type : Type::GetConfigTypesSortedByLoadDependencies()) {
+		if (auto *ctype = dynamic_cast<ConfigType *>(type.get()); ctype) {
+			for (const ConfigObject::Ptr& object : ctype->GetObjects()) {
+				// All objects must be synced sorted by their dependency graph.
+				// Otherwise, downtimes/comments etc. might get synced before their respective Checkables, which will
+				// result in comments and downtimes being ignored by the other endpoint since it does not yet know
+				// about their checkables. Given that the runtime config updates event does not trigger a reload on the
+				// remote endpoint, these objects won't be synced again until the next reload.
```

Comment:
Is UpdateConfigObjectWithParents() actually needed for dependencies _across_ types, given the added load_after?

Reply:
What would you do if the answer was no? The advantage of `DependencyGraph` is that it's supposed to track dependencies between different objects, no matter if they have the same type or not. So relying on it sounds more robust to me than mixing `load_after` and `DependencyGraph` in a way that both are necessary for correctness.

@yhabteab Though I wonder why you kept the additional `load_after` statements, i.e. do you actually expect it to fix anything in additon? I don't think they should be necessary at the moment, they should just give a roughly correct order for sending the objects to start with that will then be fixed in detail by `UpdateConfigObjectWithParents()` (which would also produce a valid order if it was called on the same objects in a random order).

Reply:
> What would you do if the answer was no?

I thought I'd rephrase the comment, but now...

> I wonder why you kept the additional `load_after` statements, i.e. do you actually expect it to fix anything in additon? I don't think they should be necessary at the moment, they should just give a roughly correct order for sending the objects to start with that will then be fixed in detail by `UpdateConfigObjectWithParents()` (which would also produce a valid order if it was called on the same objects in a random order).

... I totally agree. I'd even say we need neither load_after nor the type sorting here, given UpdateConfigObjectWithParents().

Reply:
> @yhabteab Though I wonder why you kept the additional `load_after` statements, i.e. do you actually expect it to fix anything in additon?

No, I don't expect that, but I invested some time to add them, so why would I drop them now if they don't do anything wrong. Yes, for this particular case it's enough to use the dependency graph, but I think leaving those `load_after` statements as they are now brings more advantages than disadvantages.

>  I'd even say we need neither load_after nor the type sorting here, given UpdateConfigObjectWithParents().

I didn't say it was required either, but it's already there, so why wouldn't you want to use it?